src/main/java/org/jenkinsci/plugins/deployment/HostRecords.java

```diff
@@ -0,0 +1,40 @@
+package org.jenkinsci.plugins.deployment;
+
+import hudson.EnvVars;
+import hudson.model.AbstractBuild;
+import hudson.model.EnvironmentContributingAction;
+import hudson.model.InvisibleAction;
+import org.apache.commons.lang.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class HostRecords extends InvisibleAction implements EnvironmentContributingAction {
+    Collection<HostRecord> hostrecords;
+
+    public HostRecords(Collection<HostRecord> hostRecords) {
+        this.hostrecords = hostRecords;
+    }
+
+    public Collection<HostRecord> getHostrecords() {
+        return hostrecords;
+    }
+
+    public void buildEnvVars(AbstractBuild<?, ?> abstractBuild, EnvVars envVars) {
+        List<String> deploymentHost = new ArrayList();
+        List<String> deploymentPath = new ArrayList();
+        List<String> deploymentTimeStamp = new ArrayList();
+
+        Collection<HostRecord> hostRecords = getHostrecords();
+
+        for (HostRecord hostRecord : hostRecords) {
+            deploymentHost.add(hostRecord.getHost());
+            deploymentPath.add(hostRecord.getPath());
+            deploymentTimeStamp.add(hostRecord.getTimestampString());
+        }
+        envVars.put("DEPLOYMENT_HOST", StringUtils.join(deploymentHost, ","));
+        envVars.put("DEPLOYMENT_PATH", StringUtils.join(deploymentPath, ","));
```

Comment:
Minor: there may be escaping problems in the case of weird paths. Not sure if it is really important